tls/s2n_psk.h

```diff
@@ -33,6 +33,24 @@ typedef enum {
     S2N_PSK_DHE_KE,
 } s2n_psk_key_exchange_mode;
 
+typedef enum {
+    S2N_PSK_HMAC_SHA224 = 0,
+    S2N_PSK_HMAC_SHA256,
+    S2N_PSK_HMAC_SHA384,
+} s2n_psk_hmac;
+
+struct s2n_psk_identity {
+    uint8_t *data;
+    size_t length;
+};
```

Comment:
Your change doesn't require that s2n_psk_identity be separate, so I would suggest not making it separate. The later API that needs a separate s2n_psk_identity structure can split it out :)